packages/styles/components/_c.stars-input.scss

```diff
@@ -0,0 +1,81 @@
+.mc-stars-input {
+  $parent: &;
+
+  @include set-stars-container($stars-default-size);
+
+  display: flex;
+  position: relative;
+
+  &__text,
+  &__radio {
+    @include set-hidden-input;
+  }
+
+  &__label {
+    @include set-stars-size($stars-default-size);
+
+    background:
+      url(inline-icons('star-full-32', $color-star-full)) repeat-x
+      left center;
+    border-radius: $stars-focus-radius;
+    cursor: pointer;
+
+    &::before {
+      background:
+        url(inline-icons('star-full-32', $color-star-hover)) repeat-x
+        left center;
+      content: '';
+      left: 0;
+      top: 0;
+      bottom: 0;
+      display: block;
+      position: absolute;
+      opacity: 0;
+      pointer-events: none;
+    }
+
+    @for $i from 5 through 1 {
+      &:nth-of-type(#{$i})::before {
+        width: percentage(($i - 1) / 5);
+      }
+    }
+  }
+
+  &__radio:checked ~ &__label,

```

Comment:
I kept the value `&__radio:checked ~ &__label`, because it's associated with `&__radio:focus ~ &__label`. Therefore, to keep consistency between these two selectors that apply to the same element, I kept the writing `&__radio` for both cases.